services/auth-server/app/app/views.py

```diff
@@ -85,43 +94,47 @@ def login_static(path):
         return serve_static_or_dev(path)
 
     @app.route("/login/admin", methods=["GET"])
-    def login_admin():
+    def login_admin() -> Union[Tuple[str, int], Response]:
```

Comment:
We can do

```python
from __future__ import annotations
```

and

```suggestion
    def login_admin() -> Tuple[str, int] | Response:
```

Just checked locally:

```
$ docker run -it --rm python:3.8
Python 3.8.13 (default, Aug  2 2022, 11:41:59) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from typing import Union
>>> def foo(a: Union[str, int]):
...   return a
... 
>>> def foo(a: str | int):
...   return a
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for |: 'type' and 'type'
>>> from __future__ import annotations
>>> def foo(a: str | int):
...   return a
... 
>>> foo(1)
1
```

Reply:
Overall I think I like the explicit `Union` over the syntactic sugar `|`. But that is just my opinion  :man_shrugging:

Reply:
I'm open to follow any convention the team decides. I think there's not much difference currently if we're sticking with Py3.8, because `typing` will still need to be used to import most of the library generics (i.e, `Dict`, `Tuple`, and so on...). I think the real value comes when we need to use `typing` for only a few modules like `Any` and `Literal`.

I think the team needs to also decide how to handle the import of the `typing` modules to keep the consistency in the code. Currently some modules are using `import typing as t` 

See: 
https://github.com/orchest/orchest/blob/a21bca9ac76401da2335827b7e4a7c88e1bcdec4/orchest-cli/orchestcli/cmds.py#L174-L176

Reply:
We discussed a bit internally and decided for the new syntax, also because [it's recommended in the official docs](https://docs.python.org/3/library/typing.html#typing.Union):

> To define a union, use e.g. Union[int, str] or the shorthand int | str. Using that shorthand is recommended.

About the typing imports, I'm actually fine using the `t` alias. However, I propose we do that gradually in future PRs.

For now, @sbarrios93 please use the new union syntax in the files you already changed, and this PR will be good to go 🚀